IndexError when wiring up a CZPowGate qualtran style

I'm trying to use CZPowGates as examples for some documentation and I run into an IndexError when trying to do a simple Qualtran style implementation of just a single CZPowGate:
```
from qualtran import BloqBuilder, QBit
from qualtran.bloqs.basic_gates.rotation import CZPowGate

bb = BloqBuilder()
ctrl = bb.add_register_from_dtype("ctrl", QBit())
sys = bb.add_register_from_dtype("sys", QBit())
ctrl, sys = bb.add(CZPowGate(), ctrl=ctrl, q=sys)
bb.finalize(ctrl, sys)
```
yields the error (ignoring a few intermediate frames of the stack trace):
```
[8] ctrl = bb.add_register_from_dtype("ctrl", QBit())
      [9] sys = bb.add_register_from_dtype("sys", QBit())
---> [10]ctrl, sys = bb.add(CZPowGate(), ctrl=ctrl, q=sys)
     [11] bb.finalize(ctrl, sys)
     
     ...

[976](qualtran/_infra/composite_bloq.py:976) """Add a new bloq instance to the compute graph and always return a tuple of soquets.
    [977](qualtran/_infra/composite_bloq.py:977) 
    [978](qualtran/_infra/composite_bloq.py:978) This method will always return a tuple of soquets. See `BloqBuilder.add_d(..)` for a
   (...)
...
--> [729](qualtran/_infra/composite_bloq.py:729)     idxed_soq = in_soq[li]
    [730](qualtran/_infra/composite_bloq.py:730)     assert isinstance(idxed_soq, Soquet), idxed_soq
    [731](qualtran/_infra/composite_bloq.py:731)     func(idxed_soq, reg, li)

IndexError: too many indices for array: array is 0-dimensional, but 1 were indexed
```

which is difficult to parse because as a user I have not done anything involving arrays. I can provide the intermediate frames iof the trace if necessary. When attempting to reverse engineer the previous ways in which CZPowGate is used in the library they all appear to use a cirq DecompositionContext to use the `on` method of CZPowGate as a cirq gate. One final note: there is no documentation string for CZPowGate for the user to determine what registers the bloq is supposed to act on.

hello, how to resolve this:
```
0s
1s
Run check/format-incremental
Comparing against revision 'origin/main'.
Running flynt in dry-run mode. No files will be changed.
Running the black formatter... (version: black, 24.8.0 (compiled: yes)
Python (CPython) 3.[1](https://github.com/quantumlib/Qualtran/actions/runs/15303801171/job/43060024147/#step:6:1)0.17)
--- qualtran/bloqs/basic_gates/rotation.py	2025-05-28 17:14:49.829288+00:00
+++ qualtran/bloqs/basic_gates/rotation.py	2025-05-28 17:15:59.561313+00:00
@@ -579,10 +579,[11](https://github.com/quantumlib/Qualtran/actions/runs/15303801171/job/43060024147/#step:6:12) @@
             from a discrete gate set, where a target precision `eps` is required.
 
     Registers:
         q: A single QBit register representing the qubit to be rotated.
     """
+
     angle: Union[sympy.Expr, float]
     eps: SymbolicFloat = 1e-11
 
     def decompose_bloq(self) -> 'CompositeBloq':
         raise DecomposeTypeError(f"{self} is atomic")
@@ -635,10 +636,11 @@
Error: would reformat qualtran/bloqs/basic_gates/rotation.py

Oh no! 💥 💔 💥
1 file would be reformatted.
             from a discrete gate set, where a target precision `eps` is required.
 
     Registers:
         q: A single QBit register representing the qubit to be rotated.
     """
+
     angle: Union[sympy.Expr, float]
     eps: SymbolicFloat = 1e-11
 
     def decompose_bloq(self) -> 'CompositeBloq':
         raise DecomposeTypeError(f"{self} is atomic")
Error: Process completed with exit code 1.
```

--- a/qualtran/bloqs/basic_gates/rotation.py
+++ b/qualtran/bloqs/basic_gates/rotation.py
@@ -580,6 +580,13 @@ class Rx(CirqGateAsBloqBase):
 
     Registers:
         q: A single QBit register representing the qubit to be rotated.
+
+    References:
+        [Elementary gates for quantum computation](https://arxiv.org/abs/quant-ph/9503016).
+        Barenco et al. 1995. Section 4.2 discusses single-qubit rotations.
+
+        [Quantum Computation and Quantum Information](https://www.cambridge.org/highereducation/books/quantum-computation-and-quantum-information/01E10196D0A682A6AEFFEA52D53BE9AE).
+        Nielsen and Chuang. 2010. Chapter 4.2, pp. 174-177.
     """
 
     angle: Union[sympy.Expr, float]
@@ -612,6 +619,7 @@ class Rx(CirqGateAsBloqBase):
 @frozen
 class Ry(CirqGateAsBloqBase):
     r"""Rotates a qubit about the Y-axis of the Bloch sphere.
+
     The unitary matrix for this gate is:
     $$
     R_y(\theta) = \exp(-i \frac{\theta}{2} Y) =
@@ -621,9 +629,11 @@ class Ry(CirqGateAsBloqBase):
     \end{pmatrix}
     $$
     where $\theta$ is the `angle` of rotation.
+
     This gate is equivalent to `cirq.ry(angle)`.
     It differs from `YPowGate` by a global phase. Specifically,
     `Ry(angle)` is `YPowGate(exponent=angle/np.pi, global_shift=-0.5)`.
+
     Args:
         angle: The angle of rotation in radians. This can be a symbolic expression
             or a float.
@@ -631,8 +641,16 @@ class Ry(CirqGateAsBloqBase):
             bookkeeping and does not directly affect the tensor representation
             of this gate. It becomes relevant when synthesizing rotations
             from a discrete gate set, where a target precision `eps` is required.
+
     Registers:
         q: A single QBit register representing the qubit to be rotated.
+
+    References:
+        [Elementary gates for quantum computation](https://arxiv.org/abs/quant-ph/9503016).
+        Barenco et al. 1995. Section 4.2 discusses single-qubit rotations.
+
+        [Quantum Computation and Quantum Information](https://www.cambridge.org/highereducation/books/quantum-computation-and-quantum-information/01E10196D0A682A6AEFFEA52D53BE9AE).
+        Nielsen and Chuang. 2010. Chapter 4.2, pp. 174-177.
     """
 
     angle: Union[sympy.Expr, float]

--- a/qualtran/bloqs/basic_gates/rotation.py
+++ b/qualtran/bloqs/basic_gates/rotation.py
@@ -581,6 +581,7 @@ class Rx(CirqGateAsBloqBase):
     Registers:
         q: A single QBit register representing the qubit to be rotated.
     """
+
     angle: Union[sympy.Expr, float]
     eps: SymbolicFloat = 1e-11
 
@@ -637,6 +638,7 @@ class Ry(CirqGateAsBloqBase):
     Registers:
         q: A single QBit register representing the qubit to be rotated.
     """
+    
     angle: Union[sympy.Expr, float]
     eps: SymbolicFloat = 1e-11
 

--- a/qualtran/bloqs/basic_gates/rotation.py
+++ b/qualtran/bloqs/basic_gates/rotation.py
@@ -634,7 +634,7 @@ class Ry(CirqGateAsBloqBase):
     Registers:
         q: A single QBit register representing the qubit to be rotated.
     """
-    
+
     angle: Union[sympy.Expr, float]
     eps: SymbolicFloat = 1e-11
 

--- a/qualtran/bloqs/basic_gates/rotation.py
+++ b/qualtran/bloqs/basic_gates/rotation.py
@@ -612,7 +612,6 @@ class Rx(CirqGateAsBloqBase):
 @frozen
 class Ry(CirqGateAsBloqBase):
     r"""Rotates a qubit about the Y-axis of the Bloch sphere.
-
     The unitary matrix for this gate is:
     $$
     R_y(\theta) = \exp(-i \frac{\theta}{2} Y) =
@@ -622,11 +621,9 @@ class Ry(CirqGateAsBloqBase):
     \end{pmatrix}
     $$
     where $\theta$ is the `angle` of rotation.
-
     This gate is equivalent to `cirq.ry(angle)`.
     It differs from `YPowGate` by a global phase. Specifically,
     `Ry(angle)` is `YPowGate(exponent=angle/np.pi, global_shift=-0.5)`.
-
     Args:
         angle: The angle of rotation in radians. This can be a symbolic expression
             or a float.
@@ -634,7 +631,6 @@ class Ry(CirqGateAsBloqBase):
             bookkeeping and does not directly affect the tensor representation
             of this gate. It becomes relevant when synthesizing rotations
             from a discrete gate set, where a target precision `eps` is required.
-
     Registers:
         q: A single QBit register representing the qubit to be rotated.
     """

--- a/qualtran/bloqs/basic_gates/rotation.py
+++ b/qualtran/bloqs/basic_gates/rotation.py
@@ -554,6 +554,33 @@ _CRZ_DOC = BloqDocSpec(bloq_cls=CRz, examples=[_crz])
 
 @frozen
 class Rx(CirqGateAsBloqBase):
+    r"""Rotates a qubit about the X-axis of the Bloch sphere.
+
+    The unitary matrix for this gate is:
+    $$
+    R_x(\theta) = \exp(-i \frac{\theta}{2} X) =
+    \begin{pmatrix}
+        \cos{\frac{\theta}{2}} & -i\sin{\frac{\theta}{2}} \\
+        -i\sin{\frac{\theta}{2}} & \cos{\frac{\theta}{2}}
+    \end{pmatrix}
+    $$
+    where $\theta$ is the `angle` of rotation.
+
+    This gate is equivalent to `cirq.rx(angle)`.
+    It differs from `XPowGate` by a global phase. Specifically,
+    `Rx(angle)` is `XPowGate(exponent=angle/np.pi, global_shift=-0.5)`.
+
+    Args:
+        angle: The angle of rotation in radians. This can be a symbolic expression
+            or a float.
+        eps: The precision of the rotation. This parameter is primarily for
+            bookkeeping and does not directly affect the tensor representation
+            of this gate. It becomes relevant when synthesizing rotations
+            from a discrete gate set, where a target precision `eps` is required.
+
+    Registers:
+        q: A single QBit register representing the qubit to be rotated.
+    """
     angle: Union[sympy.Expr, float]
     eps: SymbolicFloat = 1e-11
 
@@ -583,6 +610,33 @@ class Rx(CirqGateAsBloqBase):
 
 @frozen
 class Ry(CirqGateAsBloqBase):
+    r"""Rotates a qubit about the Y-axis of the Bloch sphere.
+
+    The unitary matrix for this gate is:
+    $$
+    R_y(\theta) = \exp(-i \frac{\theta}{2} Y) =
+    \begin{pmatrix}
+        \cos{\frac{\theta}{2}} & -\sin{\frac{\theta}{2}} \\
+        \sin{\frac{\theta}{2}} & \cos{\frac{\theta}{2}}
+    \end{pmatrix}
+    $$
+    where $\theta$ is the `angle` of rotation.
+
+    This gate is equivalent to `cirq.ry(angle)`.
+    It differs from `YPowGate` by a global phase. Specifically,
+    `Ry(angle)` is `YPowGate(exponent=angle/np.pi, global_shift=-0.5)`.
+
+    Args:
+        angle: The angle of rotation in radians. This can be a symbolic expression
+            or a float.
+        eps: The precision of the rotation. This parameter is primarily for
+            bookkeeping and does not directly affect the tensor representation
+            of this gate. It becomes relevant when synthesizing rotations
+            from a discrete gate set, where a target precision `eps` is required.
+
+    Registers:
+        q: A single QBit register representing the qubit to be rotated.
+    """
     angle: Union[sympy.Expr, float]
     eps: SymbolicFloat = 1e-11
 

